chao_garden/src/init.c

```diff
@@ -0,0 +1,71 @@
+#include "global.h"
+#include "agb_flash.h"
+#include "m4a.h"
+
+extern IntrFunc gIntrTable[4];
+extern u16 gUnknown_03003B70;
+
+// TCG = Tiny Chao Garden
+struct TCG_SaveSectorHeader {
+    u32 security, version;
+};
+
+struct TCG_SaveSectorData {
+    struct TCG_SaveSectorHeader header;
+
+    u8 padding_8[0x850];
+}; /* size:0x858 according to sub_02000248 */
+
+#define GBA_ROM_HEADER_FIXED_VALUE_ADDR ((vu8 *)(ROM_BASE + 0xB2))
+#define FIXED_HEADER_VALUE              0x96
+
+// TCG = Tiny Chao Garden
+#define TCG_FIRST_SAVE_SECTOR 10
+#define TCG_SAVE_SECTOR_COUNT 6
+
+extern u8 IntrMain_RAM[0x80];
+
+extern struct TCG_SaveSectorData gUnknown_03003B80;
+
+void sub_0200019c(void)
+{
+    IntrFunc *base, *table, dummy;
+
+    // Init IntrMain_RAM
+    CPU_COPY(IntrMain, IntrMain_RAM, sizeof(IntrMain_RAM), 32);
+    INTR_VECTOR = IntrMain_RAM;
+
+    // Init IntrTable
+    base = gIntrTable;
+    dummy = &IntrDummy;
+    table = base + 3;
+    do {
+        *table = dummy;
+    } while ((signed)--table >= (signed)base);
```

Comment:
Can these be `(u32)` or `(s32)`? I think it's best to specify the length (though I understand void is 4 bytes by default)

Reply:
I just wrote `signed` so that the correct conditional was used.
Not sure whether specifying the length is better or not.

Reply:
Use `(s32)` for consistency with the rest of the code base I guess

Reply:
Btw to make the comparison correct you only need to cast one side to a signed value? I think, have a go

Reply:
Nope: `src/main.c:109: warning: comparison between pointer and integer`

Reply:
Ah! Leave as both sides then